src/Microsoft.Extensions.Logging.AzureWebAppDiagnostics/Internal/BackgroundSink.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using Serilog.Core;
+using Serilog.Events;
+
+// TODO: Might want to consider using https://github.com/jezzsantos/Serilog.Sinks.Async 
+// instead of this, once that supports netstandard
+namespace Microsoft.Extensions.Logging.AzureWebAppDiagnostics.Internal
+{
+    /// <summary>
+    /// A background sink for Serilog.
+    /// </summary>
+    public class BackgroundSink : ILogEventSink, IDisposable
```

Comment:
I just spiked this up a few days ago as well: https://github.com/jezzsantos/Serilog.Sinks.Async/blob/master/src/Serilog.Sinks.Async/AsyncWorkerSink.cs - given there's a high probability of either Serilog.Sinks.Async, or eventually Serilog itself, shipping a type equivalent to this, it may be better `internal` at this stage?